azurerm/resource_arm_data_factory_trigger_schedule.go

```diff
@@ -0,0 +1,259 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/features"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmDataFactoryTriggerSchedule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmDataFactoryTriggerScheduleCreateUpdate,
+		Read:   resourceArmDataFactoryTriggerScheduleRead,
+		Update: resourceArmDataFactoryTriggerScheduleCreateUpdate,
+		Delete: resourceArmDataFactoryTriggerScheduleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAzureRMDataFactoryPipelineName,
+			},
+
+			// There's a bug in the Azure API where this is returned in lower-case
+			// BUG: https://github.com/Azure/azure-rest-api-specs/issues/5788
+			"resource_group_name": azure.SchemaResourceGroupNameDiffSuppress(),
+
+			"data_factory_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringMatch(
+					regexp.MustCompile(`^[A-Za-z0-9]+(?:-[A-Za-z0-9]+)*$`),
+					`Invalid data_factory_name, see https://docs.microsoft.com/en-us/azure/data-factory/naming-rules`,
+				),
+			},
+
+			"start_time": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				DiffSuppressFunc: suppress.RFC3339Time,
+				ValidateFunc:     validate.RFC3339Time, //times in the past just start immediately
+			},
+
+			"end_time": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				DiffSuppressFunc: suppress.RFC3339Time,
+				ValidateFunc:     validate.RFC3339Time, //times in the past just start immediately
+			},
+
+			"frequency": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  string(datafactory.Minute),
+				ValidateFunc: validation.StringInSlice([]string{
+					string(datafactory.Minute),
+					string(datafactory.Hour),
+					string(datafactory.Day),
+					string(datafactory.Week),
+					string(datafactory.Month),
+				}, false),
+			},
+
+			"interval": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntAtLeast(1),
+			},
+
+			"timezone": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "UTC",
+				ValidateFunc: validate.NoEmptyStrings,
+			},
```

Comment:
We have a few different timezone validation functions, perhaps one would work here?

Reply:
I'm hesitant to add any validation here and might actually just remove this value. It doesn't seem to do anything and the UI just defaults the time to UTC anyway

Reply:
If you pick PST the API still takes the value as UTC?

Then i would definitely remove it, add a note that the time is always UC, and open a bug on the SDK with a link in the code explaining why UTC has been ignored.

Reply:
Confirmed that timezone doesn't work no matter the value. It's been removed and an issue has been opened https://github.com/Azure/azure-sdk-for-go/issues/6244

Reply:
The Data Factory API attributes tend to accept any value without validation. You can submit anything to timezone and it'll be accepted whether it's valid or not but `Pacific Standard Time` didn't change the format of the time at all so I've submitted an issue on the sdk as a start. https://github.com/Azure/azure-sdk-for-go/issues/6244

Reply:
_fun_